### DISCUSSION TITLE
How to extract rows with duplicate values in two or more fields but different values in another field

### DISCUSSION THREAD

**Discussion by san-r** [duckdb/duckdb#2436] [Q&A]
I've posted detailed question with sample data and desired result at

https://stackoverflow.com/questions/69566587/extract-rows-with-duplicate-values-in-two-or-more-fields-but-different-values-in

If we can compose an SQL query to do the job, it would be fine. But since this is a very common operation in fraud analytics, an in-built function would be even better.

PS: I mostly work in RStudio environment and I'm trying DuckDB as a way to work with bigger-than-RAM data. My big datasets are usually kept in chunked parquet files which can be queried by DuckDB as documented at https://duckdb.org/docs/data/parquet

**Comment by jgrg** [comment]
Tersest possible DuckDB query for a table named `my_tbl` when grouping by all columns:

```sql
FROM my_tbl
GROUP BY ALL
HAVING COUNT(*) > 1;
```

Example using common table expression (the `duped` table) to select for duplicates on particular columns:

```sql
WITH raw_data(id, forename, surname, owes) AS (
  VALUES
    (1, 'Charles', 'Windsor', 300.00),
    (2, 'Philip', 'Morris', 20.40),
    (3, 'George', 'Morris', 399.99),
    (4, 'Charles', 'Windsor', 299.05)
),
duped AS (
  SELECT forename, surname
  FROM raw_data
  GROUP BY ALL
  HAVING COUNT(*) > 1
)
SELECT raw_data.*
FROM raw_data
JOIN duped USING(forename, surname);
```
